compiler/rustc_mir_dataflow/src/framework/direction.rs

```diff
@@ -12,6 +12,16 @@ pub trait Direction {
 
     const IS_BACKWARD: bool = !Self::IS_FORWARD;
 
+    fn apply_effects_in_block_and_join_state_into_successors_of<'mir, 'tcx, A>(
```

Comment:
Could you shorten the name?

Reply:
Sure. I'll just use `apply_effects_in_block`, I think that's good enough.